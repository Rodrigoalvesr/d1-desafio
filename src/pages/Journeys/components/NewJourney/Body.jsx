import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import Input from '@Components/Input';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  root: {
    marginBottom: 15,
  },
  text: {
    color: '#3E4157',
    fontSize: 13,
  },
  helper: {
    color: '#9196AB',
    fontSize: 13,
  },
});

function Body({ text, setText }) {
  const classes = useStyles();

  const handleChange = React.useCallback(
    (event) => setText(event.target.value),
    [setText]
  );

  return (
    <div className={classes.root}>
      <Typography className={classes.text}>
        Dê um <strong>nome</strong> para essa Jornada
      </Typography>
      <Input value={text} onChange={handleChange} className={classes.input} />
      <Typography className={classes.helper}>
        Você poderá alterar essa informação depois.
      </Typography>
    </div>
  );
}

Body.propTypes = {
  text: PropTypes.string.isRequired,
  setText: PropTypes.func.isRequired,
};

export default Body;
