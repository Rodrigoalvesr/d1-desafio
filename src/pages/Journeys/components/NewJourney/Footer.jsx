import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    justifyContent: 'center',
  },
});

function Footer({ handleCloseModal }) {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Button color="primary" onClick={handleCloseModal}>
        Continuar
      </Button>
      <Button onClick={handleCloseModal}>Cancelar</Button>
    </div>
  );
}

Footer.propTypes = {
  handleCloseModal: PropTypes.func.isRequired,
};

export default Footer;
