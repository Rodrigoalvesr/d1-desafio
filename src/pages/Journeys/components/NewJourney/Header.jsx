import React from 'react';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  root: {
    paddingBottom: 15,
    marginBottom: 15,
    borderBottom: '1px solid #E4E6F1',
  },
  title: {
    color: '#9196AB',
    fontSize: 14,
    fontWeight: 'bold',
  },
});

function Header() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Typography className={classes.title}>Nova Jornada</Typography>
    </div>
  );
}

export default Header;
