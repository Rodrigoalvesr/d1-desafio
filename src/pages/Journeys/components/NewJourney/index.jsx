import React from 'react';
import Modal from '@material-ui/core/Modal';
import Header from './Header';
import Body from './Body';
import Footer from './Footer';
import { useSelector, useDispatch } from 'react-redux';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  root: {
    position: 'absolute',
    width: 382,
    borderRadius: 7,
    backgroundColor: theme.palette.common.white,
    border: '1px solid #E4E6F1',
    boxShadow: '1px 1px 10px #1A173134',
    padding: 30,
    top: `50%`,
    left: `50%`,
    transform: `translate(-50%, -50%)`,
  },
}));

function NewJourney() {
  const classes = useStyles();
  const [text, setText] = React.useState('');
  const dispatch = useDispatch();
  const { modal } = useSelector((journey) => journey);

  const handleCloseModal = React.useCallback(() => {
    dispatch({ type: 'JOURNEY_CLOSE_MODAL' });
  }, [dispatch]);

  return (
    <Modal open={modal} onClose={handleCloseModal}>
      <div className={classes.root}>
        <Header />
        <Body text={text} setText={setText} />
        <Footer handleCloseModal={handleCloseModal} />
      </div>
    </Modal>
  );
}

export default NewJourney;
