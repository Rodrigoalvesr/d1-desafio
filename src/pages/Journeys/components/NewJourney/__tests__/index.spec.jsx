import React from 'react';
import renderer from 'react-test-renderer';
import NewJourney from '../index';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureMockStore([thunk]);

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useCallback: (cb) => jest.fn(cb()),
}));

describe('New journey', () => {
  it('Renders correctly', () => {
    const store = mockStore({
      modal: false,
    });
    const spy = jest.spyOn(React, 'useCallback');
    const component = renderer
      .create(
        <Provider store={store}>
          <NewJourney />
        </Provider>
      )
      .toJSON();
    expect(spy).toBeCalled();
    expect(component).toMatchSnapshot();
  });
});
