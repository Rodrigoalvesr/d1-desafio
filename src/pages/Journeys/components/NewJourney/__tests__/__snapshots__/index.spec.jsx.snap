// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New journey Renders correctly 1`] = `null`;
